core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -119,271 +130,248 @@ sealed abstract class Signer {
         BaseTxSigComponent(unsignedTx, index, spendingInfo.output, flags)
     }
   }
+
+  protected def updateScriptSigInSigComponent(
+      unsignedTx: Transaction,
+      inputIndex: Int,
+      output: TransactionOutput,
+      signatureF: Future[ScriptSignature])(
+      implicit ec: ExecutionContext): Future[BaseTxSigComponent] = {
```

Comment:
Why is this always a `BaseTxSigComponent`?

Reply:
because that is what this constructs, the two witness signers do their own funky stuff while everything else uses this so I figured I'd factor it out